### PR TITLE
Fix unused net dependency

### PR DESCRIPTION
**Pull request by niranjanyardi**
b/296715826

Remove unused net dependency

Change-Id: I8332b2b39fd2a80bc593c785ec0122791a1a74e5

### CODE REVIEW

File: cobalt/script/v8c/BUILD.gn [hunk 1]
```diff
@@ -147,6 +147,7 @@ target(final_executable_type, "v8c") {
     "//cobalt/base",
     "//cobalt/script:script",
     "//cobalt/script:standalone_javascript_runner",
+    "//net:http_server",
```
Comment (kaidokert) [inline cobalt/script/v8c/BUILD.gn:150]: This seems out of place - what pulls it in ? I wouldn't expect http_Server deps in v8c

File: cobalt/storage/BUILD.gn [hunk 2]
```diff
@@ -55,7 +55,7 @@ target(gtest_target_type, "storage_test") {
     ":storage",
     "//cobalt/base",
     "//cobalt/test:run_all_unittests",
-    "//net",
+    "//net:http_server",
```
Comment (kaidokert) [inline cobalt/storage/BUILD.gn:58]: This seems okay, it narrows the dep footprint - no action needed.

File: cobalt/storage/store/BUILD.gn [hunk 3]
```diff
@@ -40,6 +40,7 @@ target(gtest_target_type, "memory_store_test") {
     "//cobalt/base",
     "//cobalt/storage:storage_constants",
     "//cobalt/test:run_all_unittests",
+    "//net:http_server",
```
Comment (kaidokert) [inline cobalt/storage/store/BUILD.gn:43]: Again i'm surprised this is needed - which symbol / code pulls this in ?

File: cobalt/script/v8c/BUILD.gn (hunk 1 above)
Comment (niranjanyardi) [reply cobalt/script/v8c/BUILD.gn:150]: v8c somehow depends on //net which pulls in net:http_server

Comment (niranjanyardi) [reply cobalt/script/v8c/BUILD.gn:150]: Ill fix this by adding http_server as one of net dependencies

File: cobalt/storage/store/BUILD.gn (hunk 3 above)
Comment (niranjanyardi) [reply cobalt/storage/store/BUILD.gn]: memory_store the target above , pulls in net , if i change that dependcy to only http_server it screams about needing other things in net

**Pull request merged**